libraries/MenuFramework/Menu.h

```diff
@@ -0,0 +1,48 @@
+//  Created by Andrew Florjancic on 9/18/24.
+//  Copyright © 2024 Andrew Florjancic. All rights reserved
+#ifndef Menu_h
+#define Menu_h
+
+#include <Arduino.h>
+#include <DoublyLinkedList.h>
+#include "MenuItem.h"
+
+// A Menu of MenuItems that is itself a MenuItem so it can also be in a Menu.
+class Menu: public MenuItem {
+    private:
+        // An indicator that points to the current item in the menu.
+        const String cursor = "-> ";
+
+        // A list containing items that the user can select and display.
+        const DoublyLinkedList<MenuItem*>* menu;
```

Comment:
I don't think the menu itself actually needs to be a pointer. I think this is just an artifact from before addMenuItem() was created and it was expected that the main program would create the list and append items then setup the menu with the list. The MenuItems definitely need to be pointers because MenuItem is a virtual class. Let's look into this before we merge this PR.

Reply:
Updated